api/routes.py

```diff
@@ -135,15 +135,16 @@ def build_request(task_data):
 
 def get_current_user():
   """Get the username of currently logged-in user."""
-  from app import oauth
   try:
+    # Importing the oauth early results in an error
+    from app import oauth
     resp = oauth.toolhub.get("user/", token=flask.session["token"])
     resp.raise_for_status()
     profile = resp.json()
     username = profile["username"]
     return username
   except:
```

Comment:
It's best practice to avoid using generic exceptions. This will catch the case where there isn't any currently logged-in user, but also anything else that goes wrong, and we would have no way of knowing what exactly failed.

Reply:
One thing that struck me while doing this is that my error handling is very, very bad right now, particularly in the PUT request; I need to go back and sort all of this out.

Over in the data handling branch I set up the following chain of exceptions handling:
```
  except requests.exceptions.HTTPError as e:
    print("HTTP error")
    print(e.args[0])
  except requests.exceptions.ConnectionError:
    print("Connection error.  Please try again.")
  except requests.exceptions.Timeout:
    print("Request timed out.")
    # Could automatically retry
  except requests.exceptions.RequestException as e:
    print("Something went wrong.")
    print(e)
```

Leaving aside that the print() messages should be log messages (also on the todo list), is this the better way of doing things?

Reply:
It depends. If all you want to do is let the user know the exact cause of the error, just printing it out without handling each possible failure scenario separately can be enough:

```
try:
    do_the_thing()
except SomeSpecificError as spec_error:
    do_something_specific_in_this_case()
except Exception as e:
    print(e)
```

Reply:
In the case of `get_current_user`, we'd want to handle three different scenarios separately:
1. There is a logged-in user -> return username
2. There's no logged-in user -> return `None`
3. Something bad happened -> use `raise` to propagate the exception

Then the code calling `get_current_user` would do one thing or another depending on what is returned, or if an exception is raised.